qiskit/transpiler/coupling.py

```diff
@@ -403,6 +398,44 @@ def largest_connected_component(self):
         """Return a set of qubits in the largest connected component."""
         return max(rx.weakly_connected_components(self.graph), key=len)
 
+    def components(self, respect_direction: bool = False) -> List["CouplingMap"]:
+        """Separate a CouplingMap into subgraph Coupling Maps for each connected component.
+
+        This method will return a list of :class:`~.CouplingMap` objects for each connected
+        component in this :class:`~.CouplingMap`. The data payload of each node in the
+        :attr:`~.CouplingMap.graph` attribute will contain the qubit number in the original
+        graph. This will enables mapping the qubit index in a component subgraph to
+        the original qubit in the combined :class:`~.CouplingMap`. For example::
+
+            from qiskit.transpiler import CouplingMap
+
+            cmap = CouplingMap([[0, 1], [1, 2], [2, 0], [3, 4], [4, 5], [5, 3]])
+            component_cmaps = cmap.get_component_subgraphs()
+            print(component_cmaps[1].graph[0])
+
+        will print ``3`` as index ``0`` in the second component is qubit 3 in the original cmap.
+
+        Args:
+            respect_direction: If set to ``True`` the connected components will be strongly connected
+
+        Returns:
+            list: A list of :class:`~.CouplingMap` objects for each connected
+                components.
```

Comment:
There isn't, it just matches the order returned by rustworkx's connected components function (which for weakly connected components is based on a bfs starting from the lowest index). It should be deterministic, but it's not something people would be able to predict.  But, I specifically used a list here though so that it can indexed for lookup when using the return from this.

Reply:
Sure, I'll add that to the docstring. I guess that is the order, but I'd rather not commit to that as part of the API contract because it really is just an implementation detail on how rustworkx implements it's `weakly_connected_components()` function currently and not something I think we've committed to at the API level in rustworkx.

Reply:
Added in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/654a965024348799858f158c3958ae044c651700